jekyll/_cci2/using-dynamic-configuration.adoc

```diff
@@ -196,22 +198,432 @@ workflows:
+* A setup configuration, `config.yml`.
+* Separate configuration files for building the code and the docs, `code-config.yml` and `docs-config.yml`.
+* A `shared-config.yml` file that defines shared jobs to be used when building the code and the docs.
+* A `no-updates.yml` config file to be used in the event that a pipeline is triggered with no changes.
+
+Each configuration file is explained in the following sections.
+
+[,shell]
+----
+.
+├── .circleci
+│   ├── code-config.yml
+│   ├── config.yml
+│   ├── docs-config.yml
+│   ├── no-updates.yml
+│   └── shared-config.yml
+├── README.md
+├── doc
+│   └── my-docs.txt
+└── src
+    └── my-code.txt
+----
+
+[#setup-config-1]
+=== Setup configuration
+
+In this example, the setup configuration includes a single job referenced from the link:https://circleci.com/developer/orbs/orb/circleci/path-filtering#jobs-filter[path filtering orb], which is used to map pipeline parameter values and configuration files with paths to specific locations in the repository.
+
+Under the hood the following steps are taken when the pipeline is triggered:
+
+* A script runs to check for modified files in a commit against a base branch (in this example, the default, `main`).
+* A continuation configuration is generated using the relevant config files from the project, and the pipeline parameter values are set.
+* The pipeline is continued using the continuation configuration.
+
+[,yml]
+----
+version: 2.1
+
+# define this file as the setup phase of your dynamic configuration
+setup: true
+
+# define the parameters that will be used to generate the dynamic configuration.
+parameters:
+  always-continue:
+    type: boolean
+    default: false
+  build-code:
+    type: boolean
+    default: false
+  build-docs:
+    type: boolean
+    default: false
+
+# invoke the path-filtering orb to make the filter job available.
+orbs:
+  path-filtering: circleci/path-filtering@1.0.0
+
+workflows:
+  setup-workflow:
+    jobs:
+      - path-filtering/filter:
+          base-revision: main
+          config-path: .circleci/no-updates.yml
+          mapping: | # The mapping will be used to generate the dynamic configuration for all conditions that match.
+            .* always-continue true .circleci/shared-config.yml
+            src/.* build-code true .circleci/code-config.yml
+            doc/.* build-docs true .circleci/docs-config.yml
+----
+
+[#shared-jobs-1]
+=== Shared jobs
+
+In the event of changes to _any_ file in the repository (`.*`) the `shared-config.yml` configuration is included in the continuation configuration. The only time `shared-config.yml` will not be used is on a commit that contains no changes.
+
+NOTE: `shared-config.yml` is a partial configuration and contains only jobs, not workflows. This is because these jobs will be used along with either the `code-config.yml` or `docs-config.yml` configs, both of which include workflows that orchestrate these shared jobs.
```

Comment:
There are also shared parameters in the example, any part of the config that should always be available can be in here, including any workflows that should always run.